bazel/repository_locations.bzl

```diff
@@ -548,14 +548,16 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_name = "nlohmann JSON",
         project_desc = "Fast JSON parser/generator for C++",
         project_url = "https://nlohmann.github.io/json",
-        version = "3.10.4",
-        sha256 = "1155fd1a83049767360e9a120c43c578145db3204d2b309eba49fbbedd0f4ed3",
+        # 3.10.4 introduced incompatible changes with Envoy Mobile. Until Envoy Mobile updates it's
+        # minimum iOS version to 13+ this dependency cannot be updated.
+        version = "3.10.2",
```

Comment:
Wondering if the CPP 17 changes were introduced only in `3.10.4` and therefore we can rollback to `3.10.3`?

https://github.com/nlohmann/json/releases/tag/v3.10.3

Reply:
Yeah that's definitely the case. I didn't necessarily want to mess with that and have a clean revert. But if you think moving to 3.10.3 is worth it I can update @moderation

Reply:
I'm a little uncomfortable that we're stuck here. I agree we should do what is necessary to unblock Envoy Mobile and roll back (or to 3.10.3). But, we probably need some strategy around how to deal with security fixes that might arrive in the future, how long will it be until the minimum iOS version is 13+?

Reply:
> But, we probably need some strategy around how to deal with security fixes that might arrive in the future after that, how long will it be until the minimum iOS version is 13+?

It might realistically be a long time. I agree we need a better path forward here, either by:
1) Allowing upstream to move forward and overriding the old version in EM.
2) Carrying a patch in EM to remove whatever requires a newer version and/or trying to merge this patch upstream.

Reply:
Looking at the code it seems to conditionally enable the filesystem dep when supported, so it might be feasible to either include a check there that disables it for iOS or that allows passing a define that disables it?

Reply:
I wanted to do some research about Envoy Mobile's minimum iOS version before opining here.

We can, today, move to iOS 12. For 13 it'll be at least early 2022. However, there are iOS 13+ features that are of interest to Lyft so that will push us to be able to move to 13.

> so it might be feasible to either include a check there that disables it for iOS or that allows passing a define that disables it

Yes exactly. I have seen checks like this in other C++ libraries. This change can either be upstreamed to the json library itself, or as a patch in Envoy Mobile. 

That all being said, I can close this PR and leave the third party dep alone in Envoy itself. Thanks all!!